esmf_regrid/tests/unit/experimental/unstructured_scheme/test_regrid_rectilinear_to_unstructured.py

```diff
@@ -0,0 +1,125 @@
+"""Unit tests for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_rectilinear_to_unstructured`."""
+
+from iris.coords import AuxCoord, DimCoord
+from iris.cube import Cube
+import numpy as np
+
+from esmf_regrid.experimental.unstructured_scheme import (
+    regrid_rectilinear_to_unstructured,
+)
+from esmf_regrid.tests.unit.experimental.unstructured_scheme.test__cube_to_GridInfo import (
+    _grid_cube,
+)
+from esmf_regrid.tests.unit.experimental.unstructured_scheme.test__regrid_unstructured_to_rectilinear__prepare import (
+    _flat_mesh_cube,
+)
+
+
+def test_flat_cubes():
+    """
+    Basic test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_rectilinear_to_unstructured`.
+
+    Tests with flat cubes as input (a 2D grid cube and a 1D mesh cube).
+    """
+    tgt = _flat_mesh_cube()
+
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    src = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+    # Ensure data in the target grid is different to the expected data.
+    # i.e. target grid data is all zero, expected data is all one
+    tgt.data[:] = 0
+
+    def _add_metadata(cube):
+        result = cube.copy()
+        result.units = "K"
+        result.attributes = {"a": 1}
```

Comment:
Do you test that these attributes persist to the regridded object?

Reply:
You'll see that the `_add_metadata` function is applied to both `src` and `expected_cube`. I believe that testing the equality of two cubes also involves checking the equality of metadata like this, so this should all be tested by `assert expected_cube == result`.